demo/make_local_dataset_app.py

```diff
@@ -0,0 +1,214 @@
+    And save the associated text sentence in a .csv file using the same index.
+
+    Args:
+        audio_input (gr.Audio): Gradio audio object to be converted to audio data and then saved to a .wav file
+        sentence (str): The text sentence that will be associated with the audio
+        dataset_dir (str): The dataset directory path to store the indexed sentences and the associated audio files
+        index (str | None): Index of the text sentence that will be associated with the audio.
+        If None, start from 0 or append after the last element in the existing .csv
+
+    Returns:
+        str: Status text for Gradio app
+    """
+    Path(dataset_dir).mkdir(parents=True, exist_ok=True)
+    if Path(f"{dataset_dir}/text.csv").is_file():
+        text_df = pd.read_csv(f"{dataset_dir}/text.csv")
+        if index is None:
+            index = len(text_df)
+        text_df = pd.concat(
+            [text_df, pd.DataFrame([{"index": index, "sentence": sentence}])],
+            ignore_index=True,
+        )
+        text_df = text_df.drop_duplicates().reset_index(drop=True)
+    else:
+        if index is None:
+            index = 0
+        text_df = pd.DataFrame({"index": index, "sentence": [sentence]})
+
+    text_df = text_df.sort_values(by="index")
+    text_df.to_csv(f"{dataset_dir}/text.csv", index=False)
+
+    audio_filepath = f"{dataset_dir}/rec_{index}.wav"
+
+    sr, data = audio_input
+    sf.write(file=audio_filepath, data=data, samplerate=sr)
+
+    return (
+        f"""✅ Updated {dataset_dir}/text.csv \n✅ Saved recording to {audio_filepath}""",
+        None,
+    )
+
+
+def setup_gradio_demo():
+    custom_css = ".gradio-container { max-width: 450px; margin: 0 auto; }"
+    with gr.Blocks(css=custom_css) as demo:
+        gr.Markdown(
+            """
+            # 🎤 Speech-to-text Dataset Recorder
+            ### 1. Select your language
+            """
+        )
+        selected_lang = gr.Dropdown(
+            choices=list(languages_dict.keys()), value="", label="Select a language"
+        )
+        gr.Markdown(
+            "### 2. Click the Common Voice tab to use text data from the Common Voice dataset "
+            "**_OR_** Click the Custom data to write your own text data"
+        )
+        with gr.Row():
+            with gr.Tab("Common Voice"):
+                gr.Markdown("### 3. Click **Load language text dataset**.")
+                load_lang_button = gr.Button("Load Common Voice text dataset")
+                dataset_loaded = gr.Markdown()
+
+                gr.Markdown(
+                    "### 4. Set an index and click **Load from index** or use **← Previous** / **Next →** to navigate sentences."
+                )
+                with gr.Row():
+                    with gr.Column(scale=1):
+                        index = gr.Number(value=0, label="Skip to index")
+                    with gr.Column(scale=3):
```

Comment:
Is there a reason why the column doesn't use more space? In my computer everything looks squished with a lot of blank space:

![image](https://github.com/user-attachments/assets/29215ca7-b307-4955-adf3-2198ebf9732d)

Reply:
not sure whats going wrong there... mine (and stefan's last time we checked) looks like this
![Uploading image.png…]()

Reply:
ah, allright!